org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/LiveShadowedAllContainerState.java

```diff
@@ -51,14 +56,84 @@ public IAllContainersState get(IResourceDescriptions localDescriptions, IAllCont
 
 	@Override
 	public Collection<URI> getContainedURIs(String containerHandle) {
-		Set<URI> result = Sets.newLinkedHashSet();
+		Set<URI> localContainedURIs = Sets.newLinkedHashSet();
 		for (IResourceDescription descriptions : localDescriptions.getAllResourceDescriptions()) {
 			String computedHandle = getContainerHandle(descriptions.getURI());
-			if (computedHandle != null && computedHandle.equals(containerHandle))
-				result.add(descriptions.getURI());
+			if (computedHandle != null && computedHandle.equals(containerHandle)) {
+				localContainedURIs.add(descriptions.getURI());
+			}
 		}
-		result.addAll(globalState.getContainedURIs(containerHandle));
-		return result;
+		Collection<URI> globalContainedURIs = globalState.getContainedURIs(containerHandle);
+		return new Collection<URI>() {
+
+			@Override
+			public int size() {
+				return localContainedURIs.size() + globalContainedURIs.size();
```

Comment:
This will likely give a wrong answer if the URI is contained in both collection. Not sure if it matters, though.

Reply:
if i have seen correct its only used for empty calls